scripts/SANS/sans/gui_logic/presenter/run_tab_presenter.py

```diff
@@ -850,21 +853,22 @@ def _get_table_model(self):
 
         # 2. Iterate over each row, create a table row model and insert it
         number_of_rows = self._view.get_number_of_rows()
+        is_multi_period_view = self._view.is_multi_period_view()
         for row in range(number_of_rows):
-            sample_scatter = self._view.get_cell(row=row, column=SAMPLE_SCATTER_INDEX, convert_to=str)
-            sample_scatter_period = self._view.get_cell(row=row, column=SAMPLE_SCATTER_PERIOD_INDEX, convert_to=str)
-            sample_transmission = self._view.get_cell(row=row, column=SAMPLE_TRANSMISSION_INDEX, convert_to=str)
-            sample_transmission_period = self._view.get_cell(row=row, column=SAMPLE_TRANSMISSION_PERIOD_INDEX, convert_to=str)  # noqa
-            sample_direct = self._view.get_cell(row=row, column=SAMPLE_DIRECT_INDEX, convert_to=str)
-            sample_direct_period = self._view.get_cell(row=row, column=SAMPLE_DIRECT_PERIOD_INDEX, convert_to=str)
-            can_scatter = self._view.get_cell(row=row, column=CAN_SCATTER_INDEX, convert_to=str)
-            can_scatter_period = self._view.get_cell(row=row, column=CAN_SCATTER_PERIOD_INDEX, convert_to=str)
-            can_transmission = self._view.get_cell(row=row, column=CAN_TRANSMISSION_INDEX, convert_to=str)
-            can_transmission_period = self._view.get_cell(row=row, column=CAN_TRANSMISSION_PERIOD_INDEX, convert_to=str)
-            can_direct = self._view.get_cell(row=row, column=CAN_DIRECT_INDEX, convert_to=str)
-            can_direct_period = self._view.get_cell(row=row, column=CAN_DIRECT_PERIOD_INDEX, convert_to=str)
-            output_name = self._view.get_cell(row=row, column=OUTPUT_NAME_INDEX, convert_to=str)
-            user_file = self._view.get_cell(row=row, column=USER_FILE_INDEX, convert_to=str)
+            sample_scatter = self._view.get_cell(row=row, column=self.table_index['SAMPLE_SCATTER_INDEX'], convert_to=str)
+            sample_scatter_period = self._view.get_cell(row=row, column=self.table_index['SAMPLE_SCATTER_PERIOD_INDEX'], convert_to=str) if is_multi_period_view else ""  # noqa
```

Comment:
This is all super dense and pretty difficult to make sense of. I don't know if this suggestion is even worse (and you may well be able to think of something better), but could you do something like the following: 
- Wrap all the calls to `self._view.get_cell(...)` up in a nested helper:
```
def cell_value(column_name):
    return self._view.get_cell(row=row, column=self.table_index[column_name], convert_to=str)
```
- So retrieving values from the view looks much nicer:
```
sample_scatter_period = cell_value("SAMPLE_SCATTER_PERIOD_INDEX") if is_multi_period_view else ""
```
As I say, I'm not sure whether this is good practice or shoddy Pythonry so feel free to dismiss, but it would at least remove the need for those `noqa` flags

Reply:
Changed this as you suggested as I think it is better.

Reply:
Ooh this is nicer than what I suggested - nice change